cluster-provision/gocli/opts/featuregate/featuregate.go

```diff
@@ -0,0 +1,150 @@
+package featuregate
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"kubevirt.io/kubevirtci/cluster-provision/gocli/pkg/libssh"
+)
+
+type featureGateOpt struct {
+	sshClient    libssh.Client
+	featureGates []string
+}
+
+type component string
+
+const (
+	componentKubeAPIServer     component = "kube-apiserver"
+	componentKubeControllerMgr component = "kube-controller-manager"
+	componentKubeScheduler     component = "kube-scheduler"
+
+	searchFeatureGatesInFileFormat                    = "awk '/feature-gates/' %s"
+	getComponentCommandFormat                         = "kubectl --kubeconfig=/etc/kubernetes/admin.conf get pods -n kube-system -l component=%s -o jsonpath='{.items[0].spec.containers[*].command}'"
+	getComponentReadyContainersFormat                 = "kubectl --kubeconfig=/etc/kubernetes/admin.conf get pods -n kube-system -l component=%s -o=jsonpath='{.items[0].status.containerStatuses[*].ready}'"
+	addFeatureGatesToComponentCommandFormat           = `sudo sed -i 's/- %s/- %s\n    - %s/g' /etc/kubernetes/manifests/%s.yaml`
+	searchComponentsFilesCommand                      = "ls /etc/kubernetes/manifests"
+	addFeatureGatesFieldToKubeletConfigCommand        = "sudo /bin/su -c \"echo -e 'featureGates:' >> /var/lib/kubelet/config.yaml\""
```

Comment:
Using both `sudo` and `/bin/su` is redundant, can you please remove one of them?